src/victory-legend/victory-legend.js

```diff
@@ -0,0 +1,186 @@
+    symbolComponent: <Point/>,
+    labelComponent: <VictoryLabel/>,
+    containerComponent: <VictoryContainer/>,
+    groupComponent: <g/>,
+    standalone: true
+  };
+
+  getFontStyles(props) {
+    return merge({}, defaultFontStyle, props.label);
+  }
+
+  getLegendStyle(props) {
+    return {
+      style: merge({}, defaultLegendStyle, props.style)
+    };
+  }
+
+  // it should be changed to approximateTextSize
+  getLabelBbox(text, font) {
+    const body = document.body;
+    const container = document.createElement("svg");
+    const label = document.createElement("text");
+
+    label.innerText = text;
+    label.setAttribute("style",
+      `fill: ${font.color};
+      font-size: ${font.fontSize}px;
+      font-family: ${font.fontFamily};
+      background-color: ${font.backgroundColor};
+      stroke: ${font.stroke};
+      top: -10000px;
+      position: absolute;`
+    );
+    body.appendChild(container);
+    container.appendChild(label);
+
+    const bBox = label.getBoundingClientRect();
+    body.removeChild(container);
+
+    return bBox;
+  }
+
+  renderContainer(props, group) {
+    return React.cloneElement(
+      props.containerComponent,
+      this.getLegendStyle(props),
+      group
+    );
+  }
+
+  renderGroup(children) {
+    return React.cloneElement(
+      this.props.groupComponent,
+      { role: "presentation" },
+      children
+    );
+  }
+
+  render() {
+    let yPadding = 0;
+    let xPadding = 0;
+    const props = this.props;
+    const itemXPosition = [];
+    const symbolComponents = [];
+    const labelComponents = [];
+    const { symbolComponent, labelComponent } = props;
+    const data = isEmpty(props.data) ? defaultLegendData : props.data;
+
+    data.forEach((series, index) => {
+      const font = this.getFontStyles(series);
+      const symbolSize = font.fontSize;
+      const hPadding = symbolSize * 0.87;
+      const yPos = props.y + index * yPadding;
+      const labelSymbolPadding = symbolSize * 1.5 + props.columnItemSpacing;
+
+      if (props.orientation === "horizontal") {
+        itemXPosition.push(xPadding);
+        xPadding += labelSymbolPadding + this.getLabelBbox(series.name, font).width;
+      } else {
+        yPadding = symbolSize + props.rowItemSpacing;
+      }
+
+      const xLabelPos = (props.orientation !== "horizontal" ?
+        props.x : itemXPosition[index]) + hPadding;
+      const xSymbolPos = props.orientation !== "horizontal" ?
+        props.x : itemXPosition[index];
+
+      const symbolProps = assign({
+        key: `symbol-${index}`
+      }, {
+        x: xSymbolPos,
+        y: yPos,
+        size: series.symbol.size || symbolSize / 2.5,
+        symbol: series.symbol.type
+      }, series.symbol);
+
+      const labelProps = assign({
+        key: `label-${index}`
+      }, {
+        x: xLabelPos,
+        y: yPos,
+        style: font,
+        text: series.name
+      }, series.label);
+
+      symbolComponents[index] = React.cloneElement(
```

Comment:
If you move all the stuff that is actually rendered into a new method it will be much easier to write the native version.